design/one-pager-stacks-security-isolation.md

```diff
@@ -0,0 +1,239 @@
+
+### Environments
+
+While cluster stacks are installed at the cluster level (there is only one instance of each cluster stack), they can still support the concept of multiple "environments", each represented by a namespace.
+For example, an admin can create a namespace to model their production "environment" and another namespace to model their development "environment".
+All resources within each of those environments can share a common set of provider credentials.
+This namespaced "environment" approach provides the administrator with a way to collect or aggregate all the infrastructure and services associated with a given environment.
+
+#### Cluster Stack Single Instance Version Limitation
+
+One caveat of this approach is that because there is only a single instance of the cluster stack in the cluster, it is therefore shared across all environments.
+This means that all environments must use the same version of the stack.
+If an admin wants to try a newer version of the stack for their dev environment, it will also result in the newer version of the stack being used in the production environment as well.
+
+This limitation could be addressed in the future by allowing multiple versions of cluster stacks installed per control plane, which would require additional complexity and design work.
+To get started, we will make the simplifying constraint that only a single version of a cluster stack can be installed and it will be shared across all namespaces (environments).
+
+#### Environment Policies
+
+Another potential advantage of supporting multiple namespace "environments" is that each one could be assigned their own specific policy, such as default resource classes.
+The production environment could have a completely different set of default resource classes than the development environment.
+Since these environments are simply represented by namespaces, they can take on any arbitrary meaning or scope.
+For example, instead of dev vs. prod, they could represent entirely other dimensions, such as a high performance environment, a cheap environment, an AWS only environment, etc.
+
+The full design for this support of different policies per environment is out of scope of this document, but this idea has interesting potential.
+Also note that in the current implementation of policies, they must be in the same namespace as the claim referencing them.
+This will likely be updated in the future to support cluster or environment scoped policies.
+
+## Namespace Stacks
+
+Namespace Stacks are likely to represent higher level "application" functionality, but are not required to do so.
+Instead of providing the infrastructure and lower level platform components for entire environments like a cluster stack, they likely will merely provide the functionality to support a single application.
+It is expected for namespace stacks to (indirectly) consume and build on functionality from cluster stacks in the form of resource claims.
```

Comment:
Since we have a diagram of the flow, I would suggest that we link to it from here as well. It was a bit confusing reading this and trying to reason out how the namespace stacks "indirectly consume" the cluster stacks, so linking to the diagram could be helpful.